oshi-core/src/main/java/oshi/software/os/linux/LinuxUserGroupInfo.java

```diff
@@ -0,0 +1,98 @@
+package oshi.software.os.linux;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import oshi.software.os.OSUser;
+import oshi.util.ExecutingCommand;
+
+public class LinuxUserGroupInfo {
+
+	private static class UsersCache {
+		private static final long GROUPS_REFRESH_TIME_MS = 3*60*1000; //3 minutes
+		private Map<String, OSUser> usersIdMap = new HashMap<>();
+		private long lastRefreshedTimestamp;
+		
+		public OSUser getUser(String userId) {
+			if ((System.currentTimeMillis() - lastRefreshedTimestamp) > GROUPS_REFRESH_TIME_MS) {
+				refresh();
+			}
+			OSUser osUser = usersIdMap.get(userId);
+			if (osUser == null) {
+				refresh();
+				return usersIdMap.get(userId);
+			}
+			return osUser;
+		}
+		
+		private void refresh() {
+			usersIdMap.clear();
+			List<String> passwd = ExecutingCommand.runNative("getent passwd");
+			// see man 5 passwd for the fields
+			for (String entry : passwd) {
+				String[] split = entry.split(":");
+				if (split.length < 3 ){
+					continue;
+				}
+				String userName = split[0];
+				String uid = split[2];
+				// it is allowed to have multiple entries for the same userId, we use the first one
+				if (!usersIdMap.containsKey(uid)){
+					OSUser user = new OSUser();
+					user.setUserId(uid);
+					user.setUserName(userName);
+					usersIdMap.put(uid, user); 
+				}
+			}
+			lastRefreshedTimestamp = System.currentTimeMillis();
+		}
+	}
+	
+	private static class GroupsCache {
+		private static final long GROUPS_REFRESH_TIME_MS = 3*60*1000; //3 minutes
+		private Map<String, String> groupsIdMap = new HashMap<>();
+		private long lastRefreshedTimestamp;
+		
+		public String getGroup(String groupId){
+			if ((System.currentTimeMillis() - lastRefreshedTimestamp) > GROUPS_REFRESH_TIME_MS) {
+				refresh();
+			}
+			String groupName = groupsIdMap.get(groupId);
+			if (groupName == null) {
+				refresh();
+				return groupsIdMap.get(groupId);
+			}
+			return groupName;
+		}
+		
+		private void refresh() {
+			groupsIdMap.clear();
+			List<String> group = ExecutingCommand.runNative("getent group");
+			// see man 5 group for the fields
+			for (String entry : group) {
+				String[] split = entry.split(":");
+				if (split.length < 3) {
+					continue;
+				}
+				String groupName = split[0];
+				String gid = split[2];
+				if (!groupsIdMap.containsKey(gid)) {
+					groupsIdMap.put(gid, groupName);
+				}
+			}
+			lastRefreshedTimestamp = System.currentTimeMillis();
+		}
+	}
+	
+	private UsersCache usersCache = new UsersCache();
```

Comment:
Style: These two variable declarations need to be at the top of the file.  Might be worth making them final.

Reply:
Just did it. Thanks for integrating the change!